apps/backoffice/src/components/forms/controllers/multi-select-controller.tsx

```diff
@@ -27,76 +32,117 @@ export type MultiSelectControllerProps = {
+        const selected = Array.isArray(value) ? value : [];
+        const allSelected = selected.length === items.length;
+        const showDeselect = selected.length >= 2;
+        const isIndeterminate = selected.length > 0 && !allSelected;
+
+        const toggleItem = (itemValue: number | string) => {
+          const newSelected = selected.includes(String(itemValue))
+            ? selected.filter((v) => v !== String(itemValue))
+            : [...selected, String(itemValue)];
+          onChange(newSelected);
+        };
+
+        const handleSelectAll = () => {
+          onChange(allSelected ? [] : items.map((i) => String(i.value)));
+        };
+
+        const handleDeselectAll = () => {
+          onChange([]);
+        };
+
+        return (
+          <FormControl error={!!error} fullWidth margin="normal">
+            <InputLabel id={`${name}-label`} required={props.required}>
+              {props.label}
+            </InputLabel>
+            <Select
+              {...props}
+              MenuProps={{
+                PaperProps: { style: { maxHeight: 400 } },
+                disableScrollLock: true,
+              }}
+              fullWidth
+              input={<OutlinedInput label={props.label} />}
+              labelId={`${name}-label`}
+              multiple
+              onChange={() => {
+                //
+              }} // handled manually
+              renderValue={(selectedItems) => (
+                <Box sx={{ display: "flex", flexWrap: "wrap", gap: 0.5 }}>
+                  {(selectedItems as string[]).map((val) => {
+                    const label =
+                      items.find((i) => String(i.value) === val)?.label || val;
+                    return <Chip key={val} label={label} />;
+                  })}
+                </Box>
+              )}
+              value={selected}
+            >
+              {items.length > 1 && (
+                <MenuItem key="select-all" onClick={handleSelectAll}>
+                  <Checkbox
+                    checked={allSelected}
+                    indeterminate={isIndeterminate}
+                  />
+                  <ListItemText
+                    primary={labels.selectAll}
+                    sx={{ color: "#0073E6" }}
                   />
-                ))}
-              </Box>
-            )}
-            value={value ?? []}
-          >
-            {items.map((item, keyIndex) => (
-              <MenuItem key={keyIndex} value={item.value}>
-                {item.label}
-              </MenuItem>
-            ))}
-          </Select>
-          <FormHelperText error={!!error}>
-            {error ? error.message : (helperText ?? null)}
-          </FormHelperText>
-        </FormControl>
-      )}
+                </MenuItem>
+              )}
+
+              {showDeselect && (
+                <MenuItem key="deselect-all" onClick={handleDeselectAll}>
+                  <Checkbox checked={false} />
```

Comment:
The **_Select/Deselect All_** layout is not the one shown in _Figma_, which requires a single action button at a time rendered as a Text Button (without checkbox)